common/test/java/com/couchbase/lite/ReplicatorMiscTest.java

```diff
@@ -343,6 +344,78 @@ public void testDeprecatedReplicatorType() throws URISyntaxException {
         assertEquals(ReplicatorType.PULL, config.getType());
     }
 
+    /**
+     * The 4 tests below test replicator cookies option when specifying replicator configuration
+     **/
+
+    @Test
+    public void testReplicatorWithBothAuthenticationAndHeaderCookies() throws URISyntaxException {
+        Authenticator authenticator = new SessionAuthenticator("mysessionid");
+        HashMap<String, String> header = new HashMap<>();
+        header.put(AbstractCBLWebSocket.HEADER_COOKIES, "region=nw; city=sf");
+        ReplicatorConfiguration configuration = new ReplicatorConfiguration(baseTestDb, getRemoteTargetEndpoint())
+            .setAuthenticator(authenticator)
+            .setHeaders(header);
+
+        ImmutableReplicatorConfiguration immutableConfiguration = new ImmutableReplicatorConfiguration(configuration);
+        HashMap<String, Object> options = (HashMap<String, Object>) immutableConfiguration.getConnectionOptions();
+
+        assertEquals(
```

Comment:
This is very implementation dependent:  It depends on the order in which those things are concatenated into the cookie string.  I suggest you use "String.contains) instead